libpod/container_internal_linux.go

```diff
@@ -227,7 +227,7 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 			Options:     []string{"bind", "private"},
 		}
 		if c.IsReadOnly() && dstPath != "/dev/shm" {
-			newMount.Options = append(newMount.Options, "ro")
+			newMount.Options = append(newMount.Options, "ro", "nosuid", "noexec", "nodev")
```

Comment:
Does rootless bind mounts always have these flags set?  I think if any of these flags differ the mount will fail?

Reply:
it is really a runc bug here.  runc should check what flags are set and reuse them when doing `MS_REMOUNT|MS_RDONLY`.  Anyway, the issue is already tracked upstream so for now let's force these flags here, they should cause no harm.  I've tried with some bind mounts in rootless mode and they work fine